spec/controllers/articles_controller_spec.rb

```diff
@@ -0,0 +1,168 @@
+require 'rails_helper'
+
+# This spec was generated by rspec-rails when you ran the scaffold generator.
+# It demonstrates how one might use RSpec to specify the controller code that
+# was generated by Rails when you ran the scaffold generator.
+#
+# It assumes that the implementation code is generated by the rails scaffold
+# generator.  If you are using any extension libraries to generate different
+# controller code, this generated spec may or may not pass.
+#
+# It only uses APIs available in rails and/or rspec-rails.  There are a number
+# of tools you can use to make these specs even more expressive, but we're
+# sticking to rails and rspec-rails APIs to keep things simple and stable.
+#
+# Compared to earlier versions of this generator, there is very limited use of
+# stubs and message expectations in this spec.  Stubs are only used when there
+# is no simpler way to get a handle on the object needed for the example.
+# Message expectations are only used when there is no simpler way to specify
+# that an instance is receiving a specific message.
+#
+# Also compared to earlier versions of this generator, there are no longer any
+# expectations of assigns and templates rendered. These features have been
+# removed from Rails core in Rails 5, but can be added back in via the
+# `rails-controller-testing` gem.
+
+RSpec.describe ArticlesController, type: :controller do
```

Comment:
You're on the right track with this, good work so far, I'll leave a couple of examples to help you out a bit.

In general, the hierarchy you're using is fine, but rspec tests more commonly use the following:
```
describe MyClass do
  describe '#some_instance_method' do
    before do
      # some kind of test preparation that might be shared between contexts
      # for example, creating database records
    end

    context 'when it works' do
      before do
        # context specific test preparation, if needed
      end

      it 'works' do
        # your test here
      end
    end

    context 'when it does not work' do
      before do
        # context specific test preparation, if needed
      end

      it 'does not work' do
        # your test here
      end
    end
  end
end
```

So for example, you could write the `GET /index` test like this:
```
describe 'GET /index' do
  before do
    article = Article.new(valid_attributes)
    article.update(user: current_user) # this should remove the need for article.user = something and article.save
    get :index # this should work as well, but get articles_url is fine also
  end

  it 'renders a successful response' do
    expect(response).to be_successful
  end
end
```
Which is pretty much what you already have, but this way splits up the test setup from the actual test, which is the preferred thing to do in rspec.

I also see you're using `<some model>.new` or `<some model>.first_or_create` which isn't the ideal thing to do, but I suppose you didn't run into FactoryBot at this point in your Ruby journey, so that's fine as well.
If I had to guess, this is something you're going to get as a suggestion for some upcoming change, just so you have a bit of a heads up. :smile: